src/models.ts

```diff
@@ -16,19 +16,32 @@ export type TrContext = Omit<TContext, "setLocale">;
 
 type KeyPrefix<T extends string> = T extends "" ? "" : `.${T}`;
 
+interface plural {
+  one?: string;
+  zero?: string;
+  many?: string;
+};
+
+interface gender {
+  female?: string;
+  male?: string;
+};
+
 export type KeyPath<T> = (
-  T extends object
-    ? {
+  T extends plural | gender
```

Comment:
Based on our updated `DynamicSuffix` type:
```suggestion
  T extends DynamicSuffix
```